src/components/ciPipeline/SourceMaterials.tsx

```diff
@@ -164,36 +152,87 @@ export const SourceMaterials: React.FC<SourceMaterialsProps> = function (props)
+                                            </label>
+                                            <input
+                                                className="form__input"
+                                                autoComplete="off"
+                                                placeholder="Eg. main"
+                                                type="text"
+                                                disabled={!props.handleSourceChange}
+                                                value={mat.value}
+                                                onChange={(event) => {
+                                                    props?.handleSourceChange(
+                                                        event,
+                                                        mat.gitMaterialId,
+                                                        SourceTypeMap.BranchFixed,
+                                                    )
+                                                }}
+                                                autoFocus={true}
+                                            />
+                                        </div>
+                                        {props.showError && !errorObj.isValid ? (
+                                            <span className="form__error ci-error ">
+                                                <img src={error} className="form__icon" />
+                                                {props.validationRules?.sourceValue(_materials[index].value).message}
+                                            </span>
+                                        ) : (
+                                            <div className="h-18"></div>
+                                        )}
+                                    </div>
+                                )}
+
+                                {isBranchRegex && (
                                     <div className="w-50 ml-8">
                                         <label className="form__label mb-6">
-                                            {mat.type === SourceTypeMap.BranchFixed ? 'Branch Name*' : ''}
+                                            Branch Regex<span className="cr-5"> *</span>
                                         </label>
                                         <input
                                             className="form__input"
                                             autoComplete="off"
-                                            placeholder="Name"
+                                            placeholder="Enter branch regex"
                                             type="text"
                                             disabled={!props.handleSourceChange}
-                                            value={mat.value}
+                                            value={mat.regex}
                                             onChange={(event) => {
```

Comment:
same here

Reply:
it is an inline function. Can you please confirm